### PR TITLE
Show proper default destination calendar

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR fixes a bug where the incorrect calendar was being displayed as the default destination calendar. 

If we have Outlook set as the destination calendar
**Before**
image

**After**
image

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/core/CalendarManager.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export const getConnectedCalendars = async (
         if (destinationCalendar) {
           destinationCalendar.primaryEmail = primary.email;
           destinationCalendar.integrationTitle = integration.title;
+          destinationCalendar = Object.freeze(destinationCalendar);
```
Comment (joeauyeung) [inline packages/core/CalendarManager.ts:83]: This makes the destination calendar object immutable and if we do try changing the object it fails silently

Comment (zomars) [reply packages/core/CalendarManager.ts:83]: Do we know where the mutation is happening? How could we know?

Comment (joeauyeung) [reply packages/core/CalendarManager.ts:83]: I can't comment on the unchanged lines of code. This is a part of `getConnectedCalendars()` that loops through all calendar credentials and all calendars inside of that credential. The destination calendar object is declared outside of this loop. 

The bug happened when we had more than one connected account. For example, if we have three credentials and the destination calendar is in the 2nd credential. The if statement on line 80 would trigger again and change the destination calendar object.

Comment (keithwillcode) [reply packages/core/CalendarManager.ts:83]: I feel like we should leave a `TODO` (to go back at some point and find the root cause) or `HACK` comment here if we go with this solution. It's not immediately obvious to another reader of this code why the object would be frozen here nor is silently failing an ideal solution.

Comment (joeauyeung) [reply packages/core/CalendarManager.ts:84]: @keithwillcode updated with a link to my above comment. I also added a check to see if `destinationCalendar` is frozen.

**Review by emrysal — APPROVED** [review]

**Pull request merged**